2009-12-21  Chris Ball  <cjb@laptop.org>

	* jartool.c (add_file_to_jar): Test write return value against -1
	instead of 1.

--- fastjar-0.97/jartool.c.jj	2008-10-15 12:35:37.000000000 -0400
+++ fastjar-0.97/jartool.c	2009-12-22 06:48:09.309530000 -0500
@@ -1257,7 +1257,7 @@ int add_file_to_jar(int jfd, int ffd, co
       exit_on_error("write");
 
   /* write the file name to the zip file */
-  if (1 == write(jfd, fname, file_name_length))
+  if (-1 == write(jfd, fname, file_name_length))
     exit_on_error("write");
 
   if(verbose){
